src/collection.rs

```diff
@@ -0,0 +1,319 @@
+    where
+        S: Into<String>,
+    {
+        Self {
+            name: name.into(),
+            description: description.into(),
+            dtype: DataType::Float,
+            ..Default::default()
+        }
+    }
+
+    pub fn new_double<S>(name: S, description: S) -> Self
+    where
+        S: Into<String>,
+    {
+        Self {
+            name: name.into(),
+            description: description.into(),
+            dtype: DataType::Double,
+            ..Default::default()
+        }
+    }
+
+    pub fn new_string<S>(name: S, description: S) -> Self
+    where
+        S: Into<String>,
+    {
+        Self {
+            name: name.into(),
+            description: description.into(),
+            dtype: DataType::String,
+            ..Default::default()
+        }
+    }
+
+    pub fn new_varchar<S>(name: S, description: S, max_length: i32) -> Self
+    where
+        S: Into<String>,
+    {
+        assert!(max_length > 0, "max_length should be positive");
+        Self {
+            name: name.into(),
+            description: description.into(),
+            dtype: DataType::String,
+            max_length: max_length,
+            ..Default::default()
+        }
+    }
+
+    pub fn new_binary_vector<S>(name: S, description: S, dim: i32) -> Self
+    where
+        S: Into<String>,
+    {
+        assert!(dim > 0, "dim should be positive");
+        Self {
+            name: name.into(),
+            description: description.into(),
+            dtype: DataType::BinaryVector,
+            dim: dim,
+            ..Default::default()
+        }
+    }
+
+    pub fn new_float_vector<S>(name: S, description: S, dim: i32) -> Self
+    where
+        S: Into<String>,
+    {
+        assert!(dim > 0, "dim should be positive");
+        Self {
+            name: name.into(),
+            description: description.into(),
+            dtype: DataType::FloatVector,
+            dim: dim,
+            ..Default::default()
+        }
+    }
+    pub fn convert_field(self) -> schema::FieldSchema {
+        let tp = match self.dtype {
+            DataType::BinaryVector | DataType::FloatVector => vec![KeyValuePair {
+                key: "dim".to_string(),
+                value: self.dim.to_string(),
+            }],
+            DataType::VarChar => vec![KeyValuePair {
+                key: "max_length".to_string(),
+                value: self.max_length.to_string(),
+            }],
+            _ => Vec::new(),
+        };
+        schema::FieldSchema {
+            field_id: 0,
+            name: self.name,
+            is_primary_key: self.is_primary,
+            description: self.description,
+            data_type: self.dtype as i32,
+            type_params: tp,
+            index_params: Vec::new(),
+            auto_id: self.auto_id,
+        }
+    }
+}
+
+pub struct CollectionSchema {
+    inner: Vec<FieldSchema>,
+    auto_id: bool,
+}
+
+impl CollectionSchema {
+    pub fn unpack(self) -> (Vec<FieldSchema>, bool) {
```

Comment:
this moves the `inner` out, is it expected?

Reply:
`unpack` is used in `Client::convert_collection`. CollectionSchema I defined needs to unpack so that I can convert it to CollectionSchema defined in the proto file.
But now I think that moving the function `convert_collection` from `Client` to here is better, after this movement, `unpack` will be useless and I will remove it.